fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneScanAllEntriesTest.java

```diff
@@ -0,0 +1,220 @@
+            dataModel.saveRecords(1, start, context, 2);
+            if (includeEmptyDoc) {
+                LuceneIndexTestDataModel.saveEmptyRecord(isGrouped, isSynthetic, dataModel.random, dataModel.groupingKeyToPrimaryKeyToPartitionKey, start, dataModel.createOrOpenRecordStore(context), 2);
+            }
+            commit(context);
+        }
+
+        LuceneQueryClause search = matchAllDocs
+                                   ? LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY
+                                   : isSynthetic
+                                     ? new LuceneQuerySearchClause(LuceneQueryType.QUERY, CHILD_SEARCH_TERM, false)
+                                     : new LuceneQuerySearchClause(LuceneQueryType.QUERY, PARENT_SEARCH_TERM, false);
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+            LuceneScanBounds scanBounds = isGrouped
+                                          ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                          : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeyTuples(expectedResult,
+                    store.scanIndex(dataModel.index, scanBounds, null, ScanProperties.FORWARD_SCAN));
+        }
+    }
+
+    @ParameterizedTest
+    @BooleanSource
+    public void scanLargeIndexTest(boolean isGrouped) throws Exception {
+        final long seed = 6437286L;
+        final boolean isSynthetic = false;
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilder, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPrimaryKeySegmentIndexEnabled(true)
+                .setPartitionHighWatermark(10)
+                .build();
+
+        final long start = Instant.now().toEpochMilli();
+        try (FDBRecordContext context = openContext()) {
+            dataModel.saveRecords(500, start, context, 2);
+            commit(context);
+        }
+
+        // This test only uses the match-all query
+        LuceneQueryClause search = LuceneQuerySearchClause.MATCH_ALL_DOCS_QUERY;
+
+        try (FDBRecordContext context = openContext()) {
+            final FDBRecordStore store = dataModel.createOrOpenRecordStore(context);
+            LuceneScanBounds scanBounds = isGrouped
+                                          ? LuceneIndexTestValidator.groupedSortedTextSearch(store, dataModel.index, search, null, 2)
+                                          : LuceneIndexTestUtils.fullTextSearch(store, dataModel.index, search, false);
+
+            // Run the scan with the given query and assert the results
+            assertIndexEntryPrimaryKeys(l -> keyFor(2, 1001L + l), 500,
```

Comment:
Instead of regenerating, can this use `dataModel.primaryKeys(Tuple.from(2))`

Reply:
Replaced with getting the keys from the model